### ISSUE TITLE
Cannot run Qwazr in a Docker container after installing via deb

### ISSUE THREAD

**Issue by bmichalski** [qwazr/QWAZR#214]
I was trying to build a Docker container using http://download.qwazr.com/build-d907fe9/qwazr-server-1.0.0-SNAPSHOT.deb

Please find attached the Dockerfile I used : [Dockerfile.txt](https://github.com/qwazr/QWAZR/files/34105/Dockerfile.txt)

Below is the error that appears when running qwazr via

<pre>
service qwazr start
</pre>

which is the only line in /root/init.sh

Error:

<pre>
Starting QWAZR ...
/etc/init.d/qwazr: 1: cd: can't cd to /var/lib/qwazr
/etc/init.d/qwazr: 1: /etc/init.d/qwazr: cannot create /var/log/qwazr/server.out: Directory nonexistent
QWAZR started ...
</pre>

**Comment by emmanuel-keller** [comment]
While changing the name of a package, the installer used the wrong paths:
 
- /var/log/**qwazr-server**
- /var/lib/**qwazr-server**

The right ones are:

- var/log/**qwazr**
- var/lib/**qwazr**

This has been fixed in the last build:
htttp://download.qwarz.com

Can you try again ?

**Comment by emmanuel-keller** [comment]
Currently a QWAZR server can be either a master or a simple node. It can also exposes a defined set of services.

Its behaviour depends of the content of somes variable in the configuration file:
(/etc/qwazr.conf)[https://github.com/qwazr/QWAZR/blob/master/qwazr-server/src/deb/etc/qwazr.conf)

It means that it would be useful to give the user a way to choose the right behaviour.
What do you think is the best way to achieve that with Docker ?

**Comment by bmichalski** [comment]
Thank you for updating the deb package.

I created a small Dockerfile / Docker container in order to test the updated package.
See https://github.com/bmichalski-docker/docker-qwazr.

Now when I start qwazr using
service qwazr start && tail -f /var/log/qwazr/*

I get the following output 

Starting QWAZR ...
QWAZR started ...
log4j:ERROR Could not read configuration file from URL [file:/var/lib/qwazr/log4j.properties].
java.io.FileNotFoundException: /var/lib/qwazr/log4j.properties (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:557)
	at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:526)
	at org.apache.log4j.LogManager.<clinit>(LogManager.java:127)
	at org.slf4j.impl.Log4jLoggerFactory.getLogger(Log4jLoggerFactory.java:63)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:285)
	at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:305)
	at com.qwazr.utils.server.AbstractServer.<clinit>(AbstractServer.java:54)
log4j:ERROR Ignoring configuration file [file:/var/lib/qwazr/log4j.properties].
log4j:WARN No appenders could be found for logger (com.qwazr.utils.server.AbstractServer).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

Is this an expected behavior?

How can I fix it?

**Comment by bmichalski** [comment]
As for a mean to allow the configuration of the container, a user can rather easily extend / modify the Dockerfile, adding a COPY <srcFile> /etc/qwazr.conf instruction.

But that would make the configuration hardcoded.

Here's another option that comes to mind: when I was deploying https://github.com/dockerfile/elasticsearch, I liked the fact that they allow me to choose which file will be the configuration file, using a command line argument.

In the example, /elasticsearch/bin/elasticsearch -Des.config=/data/elasticsearch.yml allows me to specify that /data/elasticsearch.yml will be used as the configuration file.
As in the example, the configuration file can be mounted in the container from an external directory using -v <data-dir>:/data, therefore allowing to modify it at runtime, without having to build a new container.

Also, it could be great to make every configuration option overridable from the commandline.
Given arguments could either serve as default values, or override any option given in the file.

See https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-configuration.html and their interesting approach.

Tldr: they use a prefix to specify if the commandline argument either overrides the configuration from the file, or serves as a default value.

>Another option is to set es.default. prefix instead of es. prefix, which means the default setting will be used only if not explicitly set in the configuration file.

**Comment by emmanuel-keller** [comment]
> log4j:ERROR Could not read configuration file from URL [file:/var/lib/qwazr/log4j.properties].

Same issue for this file which was copied to /var/lib/qwazr-server. I just fixed it. The latest build is up to date now.

**Comment by bmichalski** [comment]
I tested it with [build-5e73bb6](http://download.qwazr.com/build-5e73bb6/qwazr-server-1.0.1-SNAPSHOT.deb) and I am now able to start Qwazr in a Docker container.

It seems to me this issue can be closed, thank you for the fixes.
